editor/src/components/inspector/inspector.tsx

```diff
@@ -904,3 +929,76 @@ function useBoolean(starting: boolean): {
   const toggle = React.useCallback(() => set((v) => !v), [])
   return { value, set, toggle }
 }
+
+function useShouldExpandStyleSection(): boolean {
+  const shouldExpandFromElementOrComponent = useEditorState(
+    Substores.metadata,
+    (store) =>
+      store.editor.selectedViews.every((target) => {
+        const instance = MetadataUtils.findElementByElementPath(store.editor.jsxMetadata, target)
+        if (instance == null) {
+          return true
+        }
+        return (
+          MetadataUtils.isFragmentFromMetadata(instance) ||
+          MetadataUtils.isGroupAgainstImports(instance) ||
+          isIntrinsicElementMetadata(instance)
+        )
+      }),
+    'useShouldExpandStyleSection shouldExpandFromElementOrComponent',
+  )
+
+  return useEditorState(
+    Substores.propertyControlsInfo,
+    (store) => {
+      for (const target of store.editor.selectedViews) {
```

Comment:
You've wrote a for loop here but return on the first result? Also, you could early return here if `shouldExpandFromElementOrComponent` is true

Reply:
a few weeks ago I created `mapFirstApplicable` which has the syntax of map, and runs until it gets the first non-null return value

Reply:
This has no non-null return values though

Reply:
oh

Reply:
The for loop was a mistake, I updated it to an `every` https://github.com/concrete-utopia/utopia/pull/5723/commits/f638923e97bde849807457aea6d394139e2e4fab. The intention is to bias towards collapsing the Styles section. Regarding the early return, I'd prefer to keep it because I'd have to use a fallback for the `descriptor == null || descriptor.inspector == null` case anyway